### PR TITLE
Fix design bugs

### PR DESCRIPTION
**Pull request by shucon**
### Description

Please describe your pull request.
Branching off from: https://github.com/phpmyadmin/phpmyadmin/pull/15586

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

**Comment by shucon** [comment]
List of manual tests to be done:
    * [ ]  on one table
      * [ ]  close, expand, close, expand
      * [ ]  click on browse structure
      * [ ]  move it around
      * [ ]  try to do monkey clicking and moving around
      * [ ]  click on columns names like a monkey
    * [ ]  designer
      * [ ]  click at random empty places
    * [ ]  Features
      * [ ]  Anchor, de anchor test menu
      * [ ]  Move menu from left to right
      * [ ]  Close all tables and check that the links are correctly placed, repeat the process 2+ times
      * [ ]  Enable angular links, look at the result, repeat the process 2+ times
      * [ ]  Enable grid mode, move a table and feel the grid effect, disable feel the non grid effect, repeat the process 2+ times
      * [ ]  Click help, get help in another tab and close it
      * [ ]  Monkey click on reload and have a look at the result
      * [ ]  Having display features disabled, change "shown" column, see the error message and the pink field changing places, F5 and see that the pink field did not change places
      * [ ]   Having display features disabled, change "shown" column on a already pink column, see the error message and the pink field that disapears, F5 and see that the pink field did not change places
      * [ ]   Click new table and be redirected to a new table, go back to designer
      * [ ]   Enter and quit full screen mode, repeat the process 1+ times
      * [ ]   Enter and quit full screen mode, observe the icon changing, repeat the process 1+ times
      * [ ]   Click on "show/hide" tables button, be sure it is shown, hide it, repeat the process 1+ times
      * [ ]   Click on "show/hide" tables button, observe icon changing, hide it, observe icon changing to original state, repeat the process 1+ times
      * [ ]   Save page, title must have changed to page name
      * [ ]   Delete page from previously saved page name in the list, have a success message, try another time, no more page of the name in list
      * [ ]   warning Bug to fix: page name is still as if it was saved
      * [ ]   Save page as with a new name without selecting "create a new page", ...
      * [ ]  rotating_light Save page as with a new name without selecting "create a new page", CRASH
      * [ ]    Save page as with a new name with selecting "create a new page"
      * [ ]   warning Bug to fix: page name is not changed
      * [ ]   rotating_light Bug to fix: contents are empty, all tables are gone
      * [ ]   Add new tables from another DB, title is now unsaved, save
      * [ ]   Add new tables from another DB, title is now unsaved, try adding the same tables from the same DB, a message is shown
      * [ ]   warning Bug to fix: no message
      * [ ]   Click on the ball at the end of a relation line, line is colored in red, "cancel", line is in original color
      * [ ]  New page, modal shows, accept not saving, the designer is like "reloaded"
      * [ ]  New page, modal shows, accept saving, fill the name, the designer is like "reloaded", try to delete saved page
      * [ ]  New page, modal shows, choose cancel, nothing more is done
      * [ ]  Show table list, click on  hide all, show all, repeat the process 1+ times
      * [ ]  Show table list, click on hide all, all checkbox unchecked, show all, , all checkbox checked, repeat the process 1+ times
      * [ ]  Show table list, click on hide all, icon changes, show all, icon changes, repeat the process 1+
        times
      * [ ]  Show table list, click on hide tables without relations, all tables without relations hidden, show them back, repeat the process 1+ times
      * [ ]  Show table list, click on hide tables without relations, icon changes, show them back, icon changes
      * [ ]  On table list, uncheck a table, table and relations are hidden, check the box, all is back
      * [ ]  On table list, click on structure bolt icon, you are redirected, go back to designer
      * [ ]  On table list, hide all, check on table of a relation, no relation shown but only the table, check the other table, you now have 2 tables and a relation
      * [ ]   warning Minor bug: On table list, hide all, check all back by hand, icon did not change (this is a UI bug)
      * [ ]  Table list is open, add a new table from the list, new table is listed and checked
      * [ ]  Table list is open, add a new table from the list, total tables increments by one
      * [ ]  Try hiding the new table and browsing the structure

**Review by williamdes — APPROVED** [review]
LGTM until here
Good job on the code de-duplication
Keep your commits small as possible like you already did :)

**Comment by mauriciofauth** [comment]
@shucon Could you rebase? https://github.com/phpmyadmin/phpmyadmin/pull/16331 moved the files to `js/src` and now we have to use `yarn js-compile` to compile the files with Babel.

**Comment by shucon** [comment]
Seems like I don't know how to properly rebase :cry:

**Comment by shucon** [comment]
Did it finally :tada:

**Review by williamdes — APPROVED** [review]
LGTM for now

File: js/src/designer/history.js [hunk 1]
```diff
@@ -666,17 +668,15 @@ DesignerHistory.queryFrom = function () {
     quer = '`' + temp + '`';
     tabUsed.push(temp);
 
-    // if master table (key2) matches with tab used get all keys and check if tab_left matches
-    // after this check if master table (key2) matches with tab left then check if any foreign matches with master .
-    for (i = 0; i < 2; i++) {
+    function tableMatch (a, b) {
         for (K in contr) {
             for (key in contr[K]) {// contr name
```
Comment (williamdes) [inline js/src/designer/history.js:673]: I was thinking that I did refactor the structure, did I ?

Comment (shucon) [reply js/src/designer/history.js:673]: I'm not sure. I didn't see this anywhere.

File: js/src/designer/move.js [hunk 2]
```diff
@@ -270,116 +267,122 @@ DesignerMove.resizeOsnTab = function () {
...
+    var x1Right = x1Left + designerTable.offsetWidth;
+    var x2Left  = designerForeignTable.offsetLeft;
+    var x2Right = x2Left + designerForeignTable.offsetWidth;
     var a = [];
-    var K;
-    var key;
-    var key2;
-    var key3;
-    DesignerMove.clear();
-    for (K in contr) {
```
Comment (shucon) [inline js/src/designer/move.js:287]: @williamdes we have similar code here, nice memory ;)

**Comment by shucon** [comment]
Bug: New pages are not getting saved (throws a fatal error)

File: js/src/designer/move.js [hunk 3]
```diff
@@ -1471,101 +1476,56 @@ DesignerMove.selectTab = function (t) {
...
+                    x2 + osnTab.offsetLeft,
+                    y2 - osnTab.offsetTop,
+                    'rgb(255,0,0)');
+
+                selected = 1; // Rect(x1-sm_x,y1-sm_y,10,10,"rgba(0,255,0,1)");
+                Key0 = foreignKey.tableName;
+                Key1 = foreignKey.colName;
+                Key2 = scriptContr.table.tableName;
+                Key3 = foreignKey.fkName;
+                Key = contrId;
```
Comment (shucon) [inline js/src/designer/move.js:1526]: Something wrong here, breaking relationship DELETE feature. Will check tomorrow.

Comment (shucon) [reply js/src/designer/move.js]: Fixed

File: js/src/designer/move.js [hunk 4]
```diff
@@ -594,11 +594,19 @@ DesignerMove.addOtherDbTables = function () {
         // Check if table already imported or not.
         var $table = $('[table_name="' + encodeURIComponent(db) + '.' + encodeURIComponent(table) + '"]');
         if ($table.length !== 0) {
-            Functions.ajaxShowMessage(
-                Functions.sprintf(Messages.strTableAlreadyExists, db + '.' + table),
-                undefined,
-                'error'
-            );
+            if ($table.parents()[2].style.display == 'none') {
```
Comment (shucon) [inline js/src/designer/move.js:597]: Is this fix required? If some doesn't know about `Show/hide tables list` they might get confused. @williamdes  @ibennetch 
(It's a bit hacky for now, will fix it if it's required.)

File: js/src/designer/history.js [hunk 5]
```diff
@@ -679,26 +668,25 @@ DesignerHistory.queryFrom = function () {
...
                     parts = key2.split('.');
-                    if (DesignerHistory.found(tabUsed, parts[1]) > 0) {
-                        for (key3 in contr[K][key][key2]) {
-                            parts1 = contr[K][key][key2][key3][0].split('.');
-                            if (DesignerHistory.found(tabLeft, parts1[1]) > 0) {
+                    if (DesignerHistory.found(a, parts[1]) > 0) {
+                        var contrKey2 = contr[K][key][key2];
+                        for (key3 in contrKey2) {
+                            var contrKey3 = contrKey2[key3];
+                            parts1 = contrKey3[0].split('.');
```
Comment (williamdes) [inline js/src/designer/history.js:680]: I will have to continue refactoring, this part should never exist

File: js/src/designer/move.js [hunk 6]
```diff
@@ -607,15 +619,13 @@ DesignerMove.addOtherDbTables = function () {
...
-            var tableEncoded = $($newTableDom).find('.small_tab_pref').attr('table_name_url');
-
-            if (typeof dbEncoded === 'string' && typeof tableEncoded === 'string') { // Do not try to add if attr not found !
-                $('#container-form').append($newTableDom);
-                DesignerMove.enableTableEvents(null, $newTableDom);
-                DesignerMove.addTableToTablesList(null, $newTableDom);
-                jTabs[dbEncoded + '.' + tableEncoded] = 1;
+            var dbTableNameUrl = $($newTableDom).find('.small_tab_pref').attr('unique_id');
+            if (typeof dbTableNameUrl === 'string') { // Do not try to add if attr not found !
+                // TODO: Hacky fix ($newTableDom[10])
```
Comment (williamdes) [inline js/src/designer/move.js:624]: is this one of my TODOs ?

**Review by williamdes — APPROVED** [review]
LGTM

File: js/src/designer/move.js (hunk 6 above)
Comment (shucon) [reply js/src/designer/move.js:624]: No, it's mine.

**Comment by williamdes** [comment]
Thank you for all your work !, it has been copied to #15586

**Pull request closed without merging**